tests/unit/helpers/test_search.py

```diff
@@ -26,6 +26,37 @@ def test_init():
         assert param_query[par] == add_params[par]
 
 
+def test_init_legacy_deprecation_warning():
+    with pytest.warns(
+        DeprecationWarning,
```

Comment:
Let's check for `RemovedInV4Warning`, rather than the superclass here. It's a minor difference, but since we know what the type should be, let's verify that our contract is met. We don't want to send our users bare `DeprecationWarning`s, given that we document that we emit `RemovedInV4Warning`s.